contributors/devel/pull-requests.md

```diff
@@ -1,72 +1,91 @@
+  * [5. Test](#5-test)
+  * [6. Squashing and Commit Titles](#6-squashing-and-commit-titles)
+  * [7. KISS, YAGNI, MVP, etc.](#7-kiss--yagni--mvp--etc)
+  * [8. It's OK to Push Back](#8-it-s-ok-to-push-back)
+  * [Common Sense and Courtesy](#common-sense-and-courtesy)
+<!-- END MUNGE: GENERATED_TOC -->
 
-- [Pull Request Process](#pull-request-process)
-- [Life of a Pull Request](#life-of-a-pull-request)
-  - [Before sending a pull request](#before-sending-a-pull-request)
-  - [Release Notes](#release-notes)
-    - [Reviewing pre-release notes](#reviewing-pre-release-notes)
-  - [Visual overview](#visual-overview)
-- [Other notes](#other-notes)
-- [Automation](#automation)
+# Pull Request Process - Who is this for?
 
-<!-- END MUNGE: GENERATED_TOC -->
+This doc explains the process and best practices for submitting a PR to the [Kubernetes project](https://github.com/kubernetes/kubernetes). It should serve as a reference for all contributors, and be useful especially to new and infrequent submitters.
+
+# Before You Submit a PR - Prerequisites
+
+If you're just getting started, please refer to the [development guide](development.md) to set up your local dev environment and get coding. This guide is for contributors who already have a PR to submit.
+
+**Make sure your PR adheres to our best practices. These include following project conventions, making small PRs, and commenting thoroughly. Please read the more detailed section on [Best Practices for Faster Reviews](#best-practices-for-faster-reviews) at the end of this doc.**
+
+## Run Local Verifications
+
+Run these local verifications before you submit your PR.
 
-# Pull Request Process
+* Enable, run, and pass the Kubernetes [pre-commit hook](development.md#define-a-pre-commit-hook) which checks your repo formatting (and more); note that you may want to add these when you're closer to the end of your project, as they take time to run every time you commit
+* Run and pass `make verify` (can take 30-40 minutes)
+* Run and pass `make test`
+* Run and pass `make test-integration`
 
-An overview of how pull requests are managed for kubernetes.
+## Sign the CLA
 
-This document assumes the reader has already followed the
-[development guide](development.md) to set up their environment,
-and understands
-[basic pull request mechanics](https://help.github.com/articles/using-pull-requests).
+You must sign the CLA before your first contribution. [Read more about the CLA.](https://github.com/kubernetes/community/blob/master/CLA.md)
 
+If you haven't signed the CLA before making a PR, the `k8s-ci-robot` will leave a comment with instructions on how to sign the CLA.
 
-# Life of a Pull Request
+# What to Expect When You Submit a PR
 
-Unless in the last few weeks of a milestone when we need to reduce churn and stabilize, we aim to be always accepting pull requests.
+Merging a PR requires the following steps to be completed before the PR will be merged automatically:
```

Comment:
You also go into these steps in the "The Testing and Merge Workflow" section. Perhaps link to them for more detail?